### PR TITLE
[SHOT-3836] Fixed VRED Design launching

### PR DESCRIPTION
**Pull request by rob-aitchison**
Reverted changes to the vrShotgun plugin naming

### CODE REVIEW

File: engine.py [hunk 1]
```diff
@@ -176,6 +188,27 @@ def _get_dialog_parent(self):
...
 
+    def _hide_menu_in_scripts(self):
+        """
+        Remove the entry in the VRED Scripts window
+        """
+        main_window = self._get_dialog_parent()
+        menu_names = main_window.menuBar().actions()
+
+        # Make it invisible for both Pro & Design
+        for menu_name in menu_names:
```
Comment (staceyoue) [inline engine.py:199]: Assuming VRED only has one menu action with name `"Scripts"` we can stop iterating once we've found the Scripts menu action. We could find the first occurrence like so:

```
scripts_action = next((menu_action for menu_action in menu_actions if menu_action.text() == "Scripts"), None)
if scripts_action:
    # find Shotgun menu and hide
```

We can use `next` to similarly find the Shotgun menu as well, assuming there is only one Shotgun menu

File: engine.py [hunk 2]
```diff
@@ -176,6 +188,27 @@ def _get_dialog_parent(self):
...
+                scripts_menu_items = menu_name.menu().actions()
+                for scripts_menu_item in scripts_menu_items:
+                    if scripts_menu_item.text() == "Shotgun":
+                        scripts_menu_item.setVisible(False)
+
+        # Check if it is Design and also "mute" the menu we made
+        if os.getenv("TK_VRED_EXECPATH").endswith("VREDDesign.exe"):
+            for menu_name in menu_names:
+                if menu_name.text() == "Scripts":
+                    menu_name.setVisible(False)
```
Comment (staceyoue) [inline engine.py:210]: With the suggestion above to find the `"Scripts"` action, we can avoid iterating to find the action menu a second time by now just calling:
```
if scripts_action:
    # find Shotgun menu and hide

    if os.getenv("TK_VRED_EXECPATH").endswith("VREDDesign.exe"):
        scripts_action.setVisible(False)

File: plugins/Shotgun/vrShotgun.py [hunk 3]
```diff
@@ -14,23 +14,23 @@
 
 sgtk.LogManager().initialize_base_file_handler("tk-vred")
 logger = sgtk.LogManager.get_logger(__name__)
-vrShotGrid_form, vrShotGrid_base = uiTools.loadUiType("vrShotGridGUI.ui")
+vrShotgun_form, vrShotgun_base = uiTools.loadUiType("vrShotgunGUI.ui")
```
Comment (staceyoue) [inline plugins/Shotgun/vrShotgun.py:17]: Did we mean to rename these files from ShotGrid back to Shotgun?

plugins/ShotGrid/vrShotGrid.py → plugins/Shotgun/vrShotgun.py
plugins/ShotGrid/vrShotGridGUI.ui → plugins/Shotgun/vrShotgunGUI.ui
plugins/ShotGrid/vrShotGridGUI.ui → plugins/Shotgun/vrShotgunGUI.ui

**Review by staceyoue — CHANGES_REQUESTED** [review]

File: engine.py [hunk 4]
```diff
@@ -190,24 +190,35 @@ def _get_dialog_parent(self):
...
+            None,
+        )
+        shotgun_action = next(
+            (
+                submenu_action
+                for submenu_action in scripts_action.menu().actions()
+                if submenu_action.text() == "Shotgun"
+            ),
+            None,
+        )
```
Comment (staceyoue) [inline engine.py:213]: This assumes that we will always find a `"Scripts"` menu and will error if not found.

If we want to handle this error, we should add a check to see if we found the scripts menu before trying to find the shotgun menu.

Comment (rob-aitchison) [reply engine.py]: Thanks for the thoroughness on this one @staceyoue ! It is much better because of it. 👍🏻

**Review by staceyoue — APPROVED** [review]
👍

**Pull request merged**